### DISCUSSION TITLE
Thanks...I am done!

### DISCUSSION THREAD

**Discussion by DRBAUS** [RomanKubiak/ctrlr#468] [General]
For the past few weeks this thing has been more frustration than joy.

One day things work, the other day they don't and I have reached a level that I just want to give up.

So here it is. 

Thanks everyone for your help. I am done with this. I want to use my energy for other things that are dear to me!

**Comment by damiensellier** [comment]
I feel you, sometimes it's good to take a full break on very intensive projects like those damn midi panels :)
One day, you'll get back at it hopefully and you'll find the courage to finish what you worked on so hard and for so long, this is so rewarding, when you launch a Cubase session and all your VST panels work as they should. 

As far as I'm concerned the real difficulties I encountered were not from LUA, Juce or anything Ctrlr related but buggy or unpredictible hardware, and further more the hassle of exporting VSTs in Xcode and Visual Studio, maintaining the documentation for version updates... It's a full time job.

Don't forget you can find some help here whenever you want, you'll always be welcome ;)

Enjoy your free time mate

Damien

**Reply by dnaldoog** [reply]
> I feel you, sometimes it's good to take a full break on very intensive projects like those damn midi panels :) One day, you'll get back at it hopefully and you'll find the courage to finish what you worked on so hard and for so long, this is so rewarding, when you launch a Cubase session and all your VST panels work as they should.
> 
> As far as I'm concerned the real difficulties I encountered were not from LUA, Juce or anything Ctrlr related but buggy or unpredictible hardware, and further more the hassle of exporting VSTs in Xcode and Visual Studio, maintaining the documentation for version updates... It's a full time job.
> 
> Don't forget you can find some help here whenever you want, you'll always be welcome ;)
> 
> Enjoy your free time mate
> 
> Damien

It's a very nice comment. Ctrlr can test your patience and how many hours get sucked away? Gone - you can't get 'em back -- what you hope will take a couple of weeks turns into months.

I looked at the manual for that Ensoniq BAUS and I don't think you could have chosen a more difficult task (but they are all difficult).

I hope you give it a rest and then come back to it afresh.

 A complete rewrite could also be a good idea when you come back to it. I have done that.

But, yes enjoy the break - you'll be back! 😎

**Reply by DRBAUS** [reply]
Thank you for the encouraging words!

It won't let me go. I feel it's getting very close to finishing actually and stopping would be a waste.

It is just so frustrating sometimes that one minute things work. The next the panel loads and it gives error messages.

And yes...weeks became months became years already.

**Reply by Ted-juh** [reply]
Sooner or later we all reach that boiling point. One person will persevere and the other will give up. The first option means it will eat up more of your spare time without knowing it will work anyway, the other option has taken you a lot of time for a whole lot of nothing. 

But, not just my advice, take a step back and let it rest. Give the mind some time to make way for all that you learned in the last months and come back to it afresh. But then make a schedule to not let it consume you the way it does now. Do one or two hours max a day. Progress won't be as fast but you get to enjoy more of life.

**Reply by puffer3** [reply]
This platform is poorly documented and buggy for sure, but there's nothing else like it!

I have had a hard time with the unpredictable bugs.  The help here on the forum has made it bearable.  Plus being able to control my OBsX (before only a preset oberheim poly) and GliGli Prophet 600 (lots od deep menu commands that were only accessible thru the infuriating membrane panel) in new ways has made the carrot on the stick even more appealing.  Designing the UI is also really fun for me!  

But when for some reason my project corrupts.  Or all the Lua disappears I do want to pull my hair out.

Tho if I wasn't tasked with making panels for my friend's vintage synth collection I dont think I would have ever gone down this path.  I would have just made ugly panels in Max for Live.  

It's yall helping that make the bugs and roadblocks bearable.  Thanks to this small but dedicated pool!

**Reply by tone-wreck** [reply]
Hello puffer3 and anyone who can help. I am not a developer or anything just a musician. I just got my gligli update for my old and trusted Prophet 600 and would love to be able to control it through my mac and as I understand straight from logic. But i can't seem to be able to find the panel anywhere. Could you please me point me in the right direction? Very much appreciated. Thank you for all the hard work. It is amazing to see this community in action.

**Reply by damiensellier** [reply]
Is it the one you're looking for? https://ctrlr.org/prophet-6-editor/

if it's not, just ask @Matrix12x on the gearspace.com forum he's talking about his Ctrlr pannel in the **Prophet 600 GliGli** [thread](https://gearspace.com/board/electronic-music-instruments-and-electronic-music-production/845271-prophet-600-users-have-upgrade.html).

https://gearspace.com/board/showpost.php?p=15178300&postcount=2413

https://github.com/matrix12x/p600fw/raw/master/Prophet%20600%20CTRLR%20panel%20and%20AU.zip
https://github.com/matrix12x/p600fw

**Reply by tone-wreck** [reply]
thanks a lot for this Damien. Yes i found the panel in the matrix12 thread. And also looked at the developers at imogen and son-et musique who seem to keep improving the mod constantly. It is amazing! The Prophet 600 is going to end up being the best analog synth ever... the only problem is that the panel is not updated to control the latest developments...The menu diving on the P600 is really not ideal. What a fantastic project this is.

**Reply by puffer3** [reply]
I haven't posted my finished prophet 600 panel yet. I will get around to that here soon.  It's working pretty good with the newest firmware.

**Reply by tone-wreck** [reply]
that would be amazing!

**Reply by matrix12x** [reply]
Very cool @puffer3 
Any chance you will add librarian type functionality?

**Reply by puffer3** [reply]
I am just not good enough at Lua or sysex to make a librarian work.  I will not be good enough for programing to get that working for awhile.  I do have a JSON saving method for pushing computer based presets to the synth, but its a one way system at this point.

**Reply by tone-wreck** [reply]
Hello Damien, any news on the panel? I have finally received my gligli with v2022 and I am loving it. It is really amazing,now a control panel would really make it much easier to use! I hope you are having fun completing it and so much pain...Happy hollydays.

**Reply by damiensellier** [reply]
Hi @tone-wreck, I did not work on the pro600 panel, Henry @puffer3  was doing it. Happy Holidays

**Comment by dobo365** [comment]
Hi Baus, I can understand indeed but my experience is indeed the same as Damien @damiensellier . 
It is the way the Midi implementations are done, their lack of documentation or their documentations errors that are leading to an increase of coding time.
Of course, you don't have a full good documentation of Ctrlr but with some patience and testing you can do miracles also thx to the Juce functions that are documented.
But indeed, making panels is eating most of my hobby time and almost no notes are coming out of my beloved synths.
So, I fully understand your thoughts!
Leave it for a while then maybe come back with a simpler or more limited project.
All the best...Dominique
